app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -77,6 +81,32 @@ export function* destroyPageSession({ id, withUI }) {
   }
 }
 
+function* forkPage(pageId, historyIndex) {
+  const currentPageIndex = yield select(PageSelectors.getPageIndexById, pageId);
+  const historyURLs = yield select(PageSelectors.getPageHistoryURLs, pageId);
+  const historyList = escape(JSON.stringify(historyURLs));
+  const url = `${SESSION_RESTORE_ADDR}/?history=${historyList}&historyIndex=${historyIndex}`;
+  const action = PageEffects.createPageSession(url, { selected: true });
+  const { id } = action;
+
+  yield call(createPageSession, action);
+  yield put(PageActions.setPageIndex(id, currentPageIndex + 1));
```

Comment:
I'm not aware of any cases in which pageId can't be found in displayOrder, at this time.